crates/store/re_video/src/time.rs

```diff
@@ -28,50 +28,67 @@ impl Time {
         Self(v)
     }
 
+    /// `time_base` specifies the
     #[inline]
-    pub fn from_secs(v: f64, timescale: Timescale) -> Self {
-        Self((v * timescale.0 as f64).round() as i64)
+    pub fn from_secs_since_start(
+        secs_since_start: f64,
+        timescale: Timescale,
+        start_time: Self,
+    ) -> Self {
+        Self((secs_since_start * timescale.0 as f64).round() as i64 + start_time.0)
     }
 
     #[inline]
-    pub fn from_millis(v: f64, timescale: Timescale) -> Self {
-        Self::from_secs(v / 1e3, timescale)
+    pub fn from_millis_since_start(
+        millis_since_start: f64,
+        timescale: Timescale,
+        start_time: Self,
+    ) -> Self {
+        Self::from_secs_since_start(millis_since_start / 1e3, timescale, start_time)
     }
 
     #[inline]
-    pub fn from_micros(v: f64, timescale: Timescale) -> Self {
-        Self::from_secs(v / 1e6, timescale)
+    pub fn from_micros_since_start(
+        micros_since_start: f64,
+        timescale: Timescale,
+        start_time: Self,
+    ) -> Self {
+        Self::from_secs_since_start(micros_since_start / 1e6, timescale, start_time)
     }
 
     #[inline]
-    pub fn from_nanos(v: i64, timescale: Timescale) -> Self {
-        Self::from_secs(v as f64 / 1e9, timescale)
+    pub fn from_nanos_since_start(
+        nanos_since_start: i64,
+        timescale: Timescale,
+        start_time: Self,
+    ) -> Self {
+        Self::from_secs_since_start(nanos_since_start as f64 / 1e9, timescale, start_time)
     }
 
     /// Convert to a duration
     #[inline]
     pub fn duration(self, timescale: Timescale) -> std::time::Duration {
```

Comment:
It seems like it is time to split `Time` into two types; one for time-points and one for durations. Both require `timescale`, but only one require the `start_time` offset.

Reply:
let's not explode and delay this PR even further with cosmetics like that. The time type was never meant to be all that complex, it's just a typed "this is a video time that uses a video time scale"